ios/brave-ios/Sources/BraveWallet/Crypto/Stores/NetworkStore.swift

```diff
@@ -175,19 +195,81 @@ public class NetworkStore: ObservableObject, WalletObserverStore {
     self.isSwapSupported = await swapService.isSwapSupported(chainId: chain.chainId)
   }
 
+  private func hideAllSupportedTestnets() async {
+    var networks: [BraveWallet.CoinType: [String]] = [:]
+    for chainId in WalletConstants.supportedTestNetworkChainIds {
+      var coin: BraveWallet.CoinType = .eth
+      switch chainId {
+      case BraveWallet.GoerliChainId,
+        BraveWallet.SepoliaChainId,
+        BraveWallet.FilecoinEthereumTestnetChainId:
+        coin = .eth
+      case BraveWallet.SolanaDevnet,
+        BraveWallet.SolanaTestnet:
+        coin = .sol
+      case BraveWallet.FilecoinTestnet:
+        coin = .fil
+      case BraveWallet.BitcoinTestnet:
+        coin = .btc
+      default:
+        break
+      }
+      var ids = networks[coin] ?? []
+      ids.append(chainId)
+      networks[coin] = ids
+    }
+    await rpcService.addHiddenNetworks(for: networks)
+  }
+
   @MainActor func updateChainList() async {
     // fetch all networks for all coin types
-    self.allChains = await rpcService.allNetworksForSupportedCoins(respectTestnetPreference: false)
+    self.allChains = await rpcService.allNetworksForSupportedCoins(
+      respectHiddenNetworksPreference: false
+    )
 
     // only support Ethereum custom chains
     let customChainIds = await rpcService.customNetworks(coin: .eth)
     self.customChains = allChains.filter { customChainIds.contains($0.chainId) }
+
+    // update the default network for supported coins
+    await updateDefaultNetworks()
+
+    // update hidden chains list
+    await updateHiddenChains()
+  }
+
+  @MainActor func updateDefaultNetworks() async {
+    await withTaskGroup(of: Void.self) {
+      @MainActor [weak self] group -> Void in
+      guard let self = self else { return }
+      for coinType in WalletConstants.supportedCoinTypes().elements {
+        group.addTask { @MainActor in
+          self.defaultNetworks[coinType] = await self.rpcService.network(
+            coin: coinType,
+            origin: nil
+          )
+        }
+      }
+    }
+  }
+
+  @MainActor func updateHiddenChains() async {
+    let hiddenChainIds = await rpcService.allHiddenNetworks(
+      for: WalletConstants.supportedCoinTypes().elements
+    )
+    self.hiddenChains = hiddenChainIds.compactMap({ chainId in
+      allChains.first(where: { $0.chainId == chainId })
```

Comment:
i think `chainId` comparison is enough (i can remove the `coin` comparison in the other function) + core api only return a list of hidden `chainId`s instead of `NetworkInfo`